apps/server/src/modules/user/repo/role.repo.ts

```diff
@@ -0,0 +1,98 @@
+import { EntityManager } from '@mikro-orm/mongodb';
+import { Injectable } from '@nestjs/common';
+import { EntityId, IPagination } from '@shared/domain';
+// import { QueryOrder } from '@mikro-orm/core';
+// import { Counted } from '@shared/domain/types';
+import { Role } from '../entity';
+
+interface RoleMap {
+	[name: string]: Role;
+}
+
+class RoleCache {
+	cache: RoleMap;
+
+	name: string;
+
+	constructor(name = '') {
+		this.cache = {};
+		this.name = name;
+	}
+
+	get(selector: string): Role {
+		return this.cache[selector];
+	}
+
+	has(selector: string): boolean {
+		return Object.prototype.hasOwnProperty.call(this.cache, selector);
+	}
+
+	add(selector: string, role: Role) {
+		this.cache[selector] = role;
+	}
+
+	cleanup() {
+		this.cache = {};
+	}
+}
+
+// caching ?
+@Injectable()
+export class RoleRepo {
+	cl1: RoleCache;
```

Comment:
The role repo does not need to know about what to cache. instead, a cache could be injected which is configured in a separate provider in the module definition: https://stackoverflow.com/a/62350617
Then, here we just know that we retrieve data from the caches, while we see in the module that caching is used.